micromasters/middleware.py

```diff
@@ -0,0 +1,89 @@
+"""session middleware"""
+import logging
+import time
+from importlib import import_module
+
+from django.conf import settings
+from django.contrib.sessions.backends.base import UpdateError
+from django.core.exceptions import SuspiciousOperation
+from django.utils.cache import patch_vary_headers
+from django.utils.deprecation import MiddlewareMixin
+from django.utils.http import cookie_date
+
+
+log = logging.getLogger(__name__)
+
+
+class SessionMiddleware(MiddlewareMixin):
+    """session middleware"""
+
+    def __init__(self, get_response=None):
+        super(SessionMiddleware).__init__()
+        self.get_response = get_response
+
+    def process_request(self, request):
+        """
+        loads session engine. If we are using cms then it load cache otherwise default
+        session engine from settings
+        """
+        session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME)
+        if "cms" in request.path:
+            engine = import_module("django.contrib.sessions.backends.cache")
+            log.info("session engin changes to %s", "django.contrib.sessions.backends.cache")
+        else:
+            engine = import_module(settings.SESSION_ENGINE)
+            log.info("session engin changes to %s", settings.SESSION_ENGINE)
+        request.session = engine.SessionStore(session_key)
+
+    def process_response(self, request, response):
```

Comment:
This is verbatim copied from the Django source. Why not just override the `SessionMiddleware` instead?